### PR TITLE
Support loading custom behaviors from git repo

### LINKED ISSUE #712: Support loading behaviors from a Git repo with branch + path
As a follow-up to #707, perhaps it'd be helpful to be able to specify a git repo, which the crawler would then clone locally.
Looks like we already have `git` installed in the image. The main addition would be how to specify the repo, for max flexibility.

Perhaps it could be: `--customBehaviors git+https://git.example.com/repo.git@branch#root/path` where the `@branch` and `/root/path` would be option.

Eg. given `git+https://git.example.com.git@branch#!/root/path`, the crawler would do:

```
git clone --branch branch https://git.example.com.git /tmp/behaviors-repo
cd /tmp/behaviors-repo
cd root/path
```
and then grab all of the behaviors from the current directory.

### PR DESCRIPTION
**Pull request by tw4l**
Fixes #712 

Also expands the existing documentation about behaviors and adds a test.

For the branch and relative path information I elected to go with query strings added to the URL rather than implementing a custom syntax. I think it works well but let me know what you think

### CODE REVIEW

File: tests/custom-behavior.test.js [hunk 1]
```diff
@@ -71,6 +71,41 @@ test("test mixed custom behavior sources", async () => {
   ).toBe(true);
 });
 
+test("test custom behaviors from git repo", async () => {
+  const res = child_process.execSync(
+    "docker run -v $PWD/test-crawls:/crawls webrecorder/browsertrix-crawler crawl --url https://specs.webrecorder.net/ --url https://example.org/ --url https://old.webrecorder.net/ --customBehaviors \"git+https://github.com/webrecorder/browsertrix-crawler.git?branch=custom-behaviors-testing&path=tests/custom-behaviors\" --scopeType page",
```
Comment (ikreymer) [inline tests/custom-behavior.test.js:76]: I guess we should change this to main for merge, so it uses this branch itself?

File: src/util/file_reader.ts [hunk 2]
```diff
@@ -35,6 +42,45 @@ export async function collectCustomBehaviors(
...
 
+async function collectGitBehaviors(gitUrl: string): Promise<FileSources> {
+  const url = gitUrl.split("git+").pop() || "";
+  const params = new URL(url).searchParams;
+  const branch = params.get("branch") || "";
+  const relPath = params.get("path") || "";
+  const urlStripped = url.split("?")[0];
+
+  const timestamp = Date.now();
+  const tmpDir = `/tmp/behaviors-repo-${timestamp}`;
```
Comment (ikreymer) [inline src/util/file_reader.ts:53]: I suppose very little change of conflict here, but could also use a random string, maybe for consistency

Comment (ikreymer) [inline src/util/file_reader.ts:53]:
```suggestion
  const tmpDir = `/tmp/behaviors-repo-${crypto.randomBytes(4).toString("hex")}`;
```

Maybe just generate a dir with random values, similar to other place in this file.

File: tests/custom-behavior.test.js (hunk 1 above)
Comment (tw4l) [reply tests/custom-behavior.test.js:76]: We can switch it to main, but I also created the referenced `custom-behaviors-testing` branch so as long as we don't delete that it should continue to be okay.

Comment (tw4l) [inline tests/custom-behavior.test.js:76]:
```suggestion
    "docker run -v $PWD/test-crawls:/crawls webrecorder/browsertrix-crawler crawl --url https://specs.webrecorder.net/ --url https://example.org/ --url https://old.webrecorder.net/ --customBehaviors \"git+https://github.com/webrecorder/browsertrix-crawler.git?branch=main&path=tests/custom-behaviors\" --scopeType page",
```

Comment (tw4l) [reply tests/custom-behavior.test.js]: Changed to use main to make it less brittle

**Review by ikreymer — APPROVED** [review]

**Pull request merged**